deployment/common/changeset/link_transfer_timelock_test.go

```diff
@@ -0,0 +1,153 @@
+		chainSelector: {
+			Canceller:         config,
+			Bypasser:          config,
+			Proposer:          config,
+			TimelockExecutors: []common.Address{chain.DeployerKey.From},
+			TimelockMinDelay:  big.NewInt(0),
+		},
+	})
+	require.NoError(t, env.ExistingAddresses.Merge(respTimelock.AddressBook))
+	require.NoError(t, err)
+
+	addrs, err := env.ExistingAddresses.AddressesForChain(chainSelector)
+	require.NoError(t, err)
+	require.Len(t, addrs, 5)
+
+	timelockAddress := ""
+	linkAddress := ""
+	mcmsAddress := ""
+	for key, typeAndVer := range addrs {
+		if typeAndVer.Type == types.LinkToken {
+			linkAddress = key
+		}
+		if typeAndVer.Type == types.RBACTimelock {
+			timelockAddress = key
+		}
+		if typeAndVer.Type == types.ProposerManyChainMultisig {
+			mcmsAddress = key
+		}
+	}
+
+	linkContract, err := link_token.NewLinkToken(common.HexToAddress(linkAddress), chain.Client)
+	require.NoError(t, err)
+
+	// Mint some funds
+	// grant minter permissions
+	tx, err := linkContract.GrantMintRole(chain.DeployerKey, chain.DeployerKey.From)
+	require.NoError(t, err)
+	_, err = deployment.ConfirmIfNoError(chain, tx, err)
+	require.NoError(t, err)
+
+	tx, err = linkContract.Mint(chain.DeployerKey, common.HexToAddress(timelockAddress), big.NewInt(750))
+	require.NoError(t, err)
+	_, err = deployment.ConfirmIfNoError(chain, tx, err)
+	require.NoError(t, err)
+
+	timelockContract, err := gethwrappers.NewRBACTimelock(common.HexToAddress(timelockAddress), chain.Client)
+	require.NoError(t, err)
+	timelocks := map[uint64]*gethwrappers.RBACTimelock{
+		chainSelector: timelockContract,
+	}
+	// Apply the proposal for approving token transfers with 5k allowance
+	_, err = changeset.ApplyChangesets(t, env, timelocks, []changeset.ChangesetApplication{
+		// the changeset produces proposals, ApplyChangesets will sign & execute them.
+		// in practice, signing and executing are separated processes.
+		{
+			Changeset: changeset.WrapChangeSet(changeset.LinkApproveTimelock),
+			Config: &changeset.LinkApproveTimelockRequest{
+				Allowances: []changeset.LinkAllowances{
+					{
+						Spender:   chain.DeployerKey.From,
+						Allowance: *big.NewInt(5000),
+					},
+				},
+				ChainSelector:    chainSelector,
+				LinkTokenAddress: common.HexToAddress(linkAddress),
+				TimelockAddress:  common.HexToAddress(timelockAddress),
+				MCMSAddress:      common.HexToAddress(mcmsAddress),
+				ValidUntil:       4131638958,
+				MinDelay:         0,
+			},
+		},
+	})
+
+	// Apply the changeset
+	_, err = changeset.ApplyChangesets(t, env, timelocks, []changeset.ChangesetApplication{
+		// the changeset produces proposals, ApplyChangesets will sign & execute them.
+		// in practice, signing and executing are separated processes.
+		{
+			Changeset: changeset.WrapChangeSet(changeset.LinkTransferTimelock),
+			Config: &changeset.LinkTransferTimelockRequest{
+				Transfers: []changeset.LinkTransfer{
+					{
+						To:    chain.DeployerKey.From,
```

Comment:
So it sends the LInk token to itself is it?

Reply:
no in this case the sender is not `chain.DeployerKey.From` but it is the timelock contract. So we're sending LINK from the timelock contract to EOA `chain.DeployerKey.From`

Reply:
ah got it thanks!